src/testLint/gitSecrets.test.ts

```diff
@@ -0,0 +1,127 @@
+            runCmd([gitSecrets, '--add', pattern], { cwd: toolkitProjectDir, throws: false })
+        })
+    }
+
+    function setupTestFixturesDir(toolkitProjectDir: string) {
+        const testFixturesPath = path.join(toolkitProjectDir, 'src', 'testFixtures', 'bin')
+        mkdirSync(testFixturesPath, { recursive: true })
+        return testFixturesPath
+    }
+
+    function setupAccessKeyFile(testFixturesPath: string) {
+        const accessKeyFilePath = path.join(testFixturesPath, 'fileWithAccessKey.ts')
+        deleteFileIfExists(accessKeyFilePath)
+        return accessKeyFilePath
+    }
+
+    function setupGitSecretsExecutable(testFixturesPath: string) {
+        const gitSecretsExecutablePath = path.join(testFixturesPath, 'git-secrets')
+
+        if (existsSync(gitSecretsExecutablePath)) {
+            console.log('INFO: git-secrets already installed')
+        } else {
+            console.log('INFO: Installing git-secrets...')
+            runCmd(['mkdir', '-p', path.parse(gitSecretsExecutablePath).dir])
+            runCmd([
+                'curl',
+                '-o',
+                gitSecretsExecutablePath,
+                'https://raw.githubusercontent.com/awslabs/git-secrets/99d01d58ebcc06e237c0e3f3ff5ae628aeef6aa6/git-secrets',
+            ])
+            runCmd(['chmod', '+x', gitSecretsExecutablePath])
+        }
+
+        return gitSecretsExecutablePath
+    }
+
+    function deleteFileIfExists(filePath: string) {
+        if (existsSync(filePath)) {
+            unlinkSync(filePath)
+        }
+    }
+
+    function createFileWithSecretKey(accessKeyFilePath: string) {
+        // Create file in project that has secret key value.
+        // Need to build access key string incrementally to not trigger git-secrets.
+        const keyValue = 'yAki21XLhAIBiKvyaxr4p/ltr8OxkZTHISISFAKE'
+        const mySecretAccessKey = `const x = { "aws_secret_access_key": "${keyValue}" }`
+        const fileContent = `
+/*!
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+${mySecretAccessKey}
+`.trim()
+        writeFileSync(accessKeyFilePath, fileContent)
+    }
+
+    before(function () {
+        if (platform() === 'win32') {
+            this.skip()
+        }
+
+        toolkitProjectDir = path.resolve()
+        testFixturesPath = setupTestFixturesDir(toolkitProjectDir)
+        gitSecrets = setupGitSecretsExecutable(testFixturesPath)
+        accessKeyFilePath = setupAccessKeyFile(testFixturesPath)
+
+        // Register all patterns with `git-secrets`
+        runCmd([gitSecrets, '--register-aws'], { cwd: toolkitProjectDir })
+        setDenyListPatterns(gitSecrets)
+        setAllowListPatterns(gitSecrets)
+    })
+
+    afterEach(function () {
+        deleteFileIfExists(accessKeyFilePath)
+    })
+
+    it('ensures no git secrets are found', function () {
+        const result = runCmd([gitSecrets, '--scan'], { cwd: toolkitProjectDir })
```

Comment:
✔ ensures no git secrets are found (68ms)

68ms to scan the whole codebase is fast, so I guess we don't need to pass the "diff".